File: server/UserData/Item/item.cs
using Nanina.UserData.ModifierData;

namespace Nanina.UserData.ItemData
{
    public enum ItemType 
    {
        Equipment,
        UserConsumable,
        WaifuConsumable,
        Material,
    }
    public class Item
    {
        public short id {get; set;}
        public int inventoryId {get; set;}
        public int count {get; set;}
        public ItemType type {get; set;}
        public required string imgPATH {get; set;}
        public short rarity {get; set;}
        public required List<Modifier> modifiers {get; set;}
    }
}

File: client/src/classes/item/item.ts
import ItemType from "./item_type";
import Modifier from '../modifiers/modifiers'

export default class Item {
    public inventoryId! : number
    public count : number = 1;
    public id : number = 0;
    public imgPATH : string = "img path";
    public name : string = "item name";
    public description : string = "item description";
    public type : ItemType = ItemType.Material
    public modifiers : Modifier[] = []
    public rarity : number = 1
}